### ISSUE TITLE
Loading multiple MP4 files in parallel causes problems with trying to add existing key to dictionary

### ISSUE THREAD

**Issue by Pedrokostam** [Zeugma440/atldotnet#124]
## The problem
I was trying to speed up loading the files by doing it in multiple parallel tasks. My console app started to print some stack traces telling informing me, that the app was trying to add an existing key to a dictionary. [stack_trace.txt](https://github.com/Zeugma440/atldotnet/files/7833226/stack.trace.txt)

## Environment

* Windows 11
* Nuget package 4.01
* Commit f39fde9441b94ec86a5ab7f2645c9ab6f1783fed

## Details
I checked though the source and it seems that method `addFrameClass(string, byte)` in `ATL.AudioData.IO.MP4` is the culprit. It checks if a static dictionary contains a frame code key and adds it if missing. Since it is static it is possible that between the checking and adding, another thread will have already added it. It causes an exception which is caught in `ReadFromFile(bool, bool)` in `ATL.AudioData.AudioDataManager` and the stack trace is forwarded to console.

I don't know if the problem occurs with other formats.

## Code To Reproduce Issue

I attached a project that can replicate the problem by loading a custom MP4 file (tag structure copied from an iTunes release) in multiple tasks. It does not always replicate the issue, though.
[Multithreading.zip](https://github.com/Zeugma440/atldotnet/files/7833216/Multithreading.zip)

**Comment by Zeugma440** [comment]
Thanks for your report. At that point, the library isn't exactly concurrency-proof, so these bug reports are welcome 😅 

I'll get back to you when the fix is avalable.

**Comment by Zeugma440** [comment]
The problem should be fixed in the next update. I'm waiting for the feedback from another issue to publish it.

**Comment by Zeugma440** [comment]
Included in today's v4.02. Please reopen this issue if you have another concurrency issue
